azure/ConsiderationsForServiceDesign.md

```diff
@@ -141,94 +144,205 @@ cannot collide with a resource path that contains user-specified resource ids.
 Long-running operations are an API design pattern that should be used when the processing of
 an operation may take a significant amount of time -- longer than a client will want to block
 waiting for the result.
-Azure allows for two forms of this design pattern: resource-based long-running operations (RELO),
-which is the preferred pattern, and long-running operations with a status monitor.
 
-In both patterns, the processing of the operation is initiated by one API call and the client
-obtains the results of the operation from a subsequent API call.
-Here we illustrate the sequence of API calls involved in each of these patterns.
+The request that initiates a long-running operation returns a response that points to or embeds
+a _status monitor_, which is an ephemeral resource that will track the status and final result of the operation.
+The status monitor resource is distinct from the target resource (if any) and specific to the individual
+operation request.
+
+A POST or DELETE operation returns a `202 Accepted` response with the status monitor in the response body.
+A long-running POST should not be used for resource create -- use PUT as described below.
+PATCH must never be used for LROs -- it should be reserved for simple resource updates.
+If a long-running update is required it should be implemented with POST.
+
+There is a special form of long-running operation initiated with PUT that is described
+in [Create (PUT) with additional long-running processing](#create-put-with-additional-long-running-processing).
+The remainder of this section describes the pattern for long-running POST and DELETE operations.
+
+This diagram illustrates how a long-running operation with a status monitor is initiated and then how the client
+determines it has completed and obtains its results:
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant API Endpoint
+    participant Status Monitor
+    Client->>API Endpoint: POST/DELETE
+    API Endpoint->>Client: HTTP/1.1 202 Accepted<br/>Retry-After: 5<br/>{ "id": "22", "status": "NotStarted" }
+    Client->>Status Monitor: GET
+    Status Monitor->>Client: HTTP/1.1 200 Ok<br/>Retry-After: 5<br/>{ "id": "22", "status": "Running" }
+    Client->>Status Monitor: GET
+    Status Monitor->>Client: HTTP/1.1 200 Ok<br/>{ "id": "22", "status": "Succeeded" }
+```
 
-### Resource-based long-running operations
+1. The client sends the request to initiate the long-running operation.
+The initial request could be a PUT, POST, or DELETE method.
+The request may contain an `operation-id` header that the service uses as the ID of the status monitor created for the operation.
```

Comment:
Should we specify what is a valid `operation-id`? I guess we don't want user to inject an URL to service (the status monitor URL).

Is it "MUST" in POST? If we lost the response, without it, we seems not able to get the result.

Reply:
We have debated the optional vs required for `operation-id` and (maybe grudgingly) decided on optional. We could reopen the discussion ...

Reply:
No, we don't need to reopen the discussion (yes, remember discussion on whether to merge it with `Repeatability-Request-ID`.

What I think to be important is the be precise what value of `operation-id` is acceptable. This way, both user and backend developer would be clear on what to send and what to expect (I assume user cannot send `!@#$%^&*(` as `operation-id`?).

Reply:
I see there is some wording around an operation-id that it can be a GUID in the Guidelines.md file

Reply:
When we update the Azure API Style guide I think we can say that the operation-id header should document allowed characters (with `pattern`) and maximum length, which is already documented for ids passed in path parameters.

https://github.com/Azure/azure-api-style-guide/blob/main/openapi-style-guide.md#path-parameters

Reply:
OK.

Though I would suggest that we say here that GUID "must" be supported, whatever other pattern is allowed.
This would give user the minimal guarantee that if they use a GUID, it will be accepted, whatever service one is communicating (another is to use the same value of `Repeatability-Request-ID`).

Reply:
I think anything other than using a GUID is inviting trouble since most well behaved services would use a GUID (or possibly some form of hash for the request - but see below) in order to avoid name conflicts (which are really bad for operations). The GUID-to-string conversion is sensitive to a couple of different things; with-or-without curly braces, with-or-without dashes and case sensitivity.

Reply:
> 

I believe there are two aspects here:

1) The 'operation-id' being recommended and the edge cases covered here. 
2) The 'operation-id' being a type that's in-depth covered in the style guideline with recommendations around the normalization and code generation